packages/quantic/cypress/e2e/default-2/search-box/search-box.cypress.ts

```diff
@@ -0,0 +1,738 @@
+              Expect.displayClearRecentQueriesButton(true);
+              Expect.numberOfQuerySuggestions(expectedQuerySuggestions.length);
+              Expect.querySuggestionsEquals(expectedQuerySuggestions);
+            });
+
+            scope('when selecting a recent query with the keyboard', () => {
+              // First arrow down press to go to the clear recent queries option.
+              Actions.pressDownArrowOnSearchBox(textarea);
+              // pressing the arrow down button to surpass every recent query option.
+              for (let i = 0; i < exampleRecentQueries.length + 1; i++) {
+                Actions.pressDownArrowOnSearchBox(textarea);
+              }
+
+              Actions.pressEnterOnSearchBox(textarea);
+              const clickedSuggestionIndex = 0;
+              Expect.searchWithQuery(
+                exampleQuerySuggestions[clickedSuggestionIndex],
+                {
+                  LSkey: recentQueriesLSKey,
+                  queries: [
+                    exampleQuerySuggestions[clickedSuggestionIndex],
+                    ...exampleRecentQueries,
+                  ],
+                }
+              );
+              Expect.logClickSuggestion(
+                exampleQuerySuggestions[clickedSuggestionIndex]
+              );
+            });
+          });
+        });
+
+        describe('when a custom value is set for the property #numberOfSuggestions', () => {
+          const exampleRecentQueries = ['foo', 'bar'];
+          const customNumberOfSuggestions = 3;
+
+          beforeEach(() => {
+            setRecentQueriesInLocalStorage(exampleRecentQueries);
+            visitSearchBox({
+              ...defaultOptions,
+              textarea,
+              numberOfSuggestions: customNumberOfSuggestions,
+            });
+            mockQuerySuggestions(exampleQuerySuggestions);
+          });
+
+          it('should limit the number of suggestions to display to respect the number of suggestions property', () => {
+            const expectedQuerySuggestions = [
+              ...exampleRecentQueries,
+              ...exampleQuerySuggestions,
+            ].slice(0, customNumberOfSuggestions);
+            scope('when loading standalone search box', () => {
+              Expect.displaySearchBoxInput(true, textarea);
+              Expect.displaySearchButton(true);
+            });
+
+            scope('when focusing on the search box input', () => {
+              Actions.focusSearchBox(textarea);
+              cy.wait(InterceptAliases.QuerySuggestions);
+
+              Expect.displaySuggestionList(true);
+              Expect.displayClearRecentQueriesButton(true);
+              Expect.numberOfQuerySuggestions(customNumberOfSuggestions);
+              Expect.querySuggestionsEquals(expectedQuerySuggestions);
+            });
+          });
+        });
+
+        describe('when a the property #disableRecentQueries is set to true', () => {
```

Comment:
The only thing we should potentially test with e2e here, 

we should execute a query, and check that it's not showing up in the list when focusing on the searchbox.